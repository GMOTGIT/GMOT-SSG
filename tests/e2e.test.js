const runSSG = require("./runSSG"); 

describe("End to End testing", () => {
    test("should print usage and options when --help option is specified", async () => {
        const { stderr, stdout, exitCode} = await runSSG("--help");

        expect(exitCode).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual("");    
    });
    test("should print nothing when the program runs succesfully", async () => {
        const { stderr, stdout, exitCode } = await runSSG("-i", "test.txt");

        expect(exitCode).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual("");    
    });
    test("should print the version of the program when --version is specified", async () => {
        const { stderr, stdout, exitCode } = await runSSG("--version");

        expect(exitCode).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual("");    
    });
})