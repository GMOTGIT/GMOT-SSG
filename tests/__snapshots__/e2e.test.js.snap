// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End to End testing should print nothing when the program runs succesfully 1`] = `""`;

exports[`End to End testing should print the version of the program when --version is specified 1`] = `"1.0.0"`;

exports[`End to End testing should print usage and options when --help option is specified 1`] = `
"Usage: GMOT-SSG.js <command> [file or directory]

Options:
  -i, --input       Load a file or a directory                           [array]
  -h, --help        Show help                                          [boolean]
  -o, --output      Set a directory for output
  -s, --stylesheet  Set a stylesheet for the template
  -c, --config      Receive a Config File and read input from it
  -v, --version     Show version number                                [boolean]

Examples:
  GMOT-SSG.js SSG -i myExample.txt          myExample.html and
  directoryPath                             txtFromDirectory.html is created

copyright 2021"
`;
